src/internal/build.ts

```diff
@@ -4,7 +4,7 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import {Deferred} from '../shared/deferred';
+import {Deferred} from '../shared/deferred.js';
```

Comment:
Great fix! Is there a way to programmatically check for this in the future, i.e. lint pass or a test webpack build?

Reply:
Good idea. Set up eslint with https://github.com/import-js/eslint-plugin-import/blob/master/docs/rules/extensions.md. Fixed all the other issues eslint found too.